.github/workflows/build_and_test.yml

```diff
@@ -168,8 +170,6 @@ jobs:
           echo "CLIPPY_COMMON_ARGS=${CLIPPY_COMMON_ARGS}" >> $GITHUB_ENV
       - name: Run cargo clippy (debug)
         run: cargo hack --feature-powerset clippy $CLIPPY_COMMON_ARGS
-      - name: Run cargo clippy (release)
-        run: cargo hack --feature-powerset clippy --release $CLIPPY_COMMON_ARGS
```

Comment:
Taking into account my previous comment (and if my assumption is correct), we have 2 build types to catch possible differences from `#[cfg(debug_assertions)]` macro

Reply:
Still seems very heavy-handed to compile the whole thing twice, for a few differing lines. Is it worth it?

Or can we make the second run cheaper? Could we run clippy in mode, but with debug_assertions=false so that clippy covers those codepaths but doesn't spend time on the extra optimizations?

Reply:
>  Yep, I think we use -C debug-assertions=no instead of the full release build.

Do you mean, "we [can] use -C debug-assertions=no ... "  ? Ok, I'll do that, let's see how much faster it is.

Reply:
Yep, I've pushed it in https://github.com/neondatabase/neon/pull/8912/commits/f25e07893cab3fa75473af639c6f5626ca67b589, got ~1m speed-up